distutils/dist.py

```diff
@@ -346,7 +346,12 @@ def _gen_paths(self):
         prefix = '.' * (os.name == 'posix')
         filename = prefix + 'pydistutils.cfg'
         if self.want_user_cfg:
-            yield pathlib.Path('~').expanduser() / filename
+            try:
+                yield pathlib.Path('~').expanduser() / filename
+            except RuntimeError:
+                warnings.warn(
```

Comment:
Diffcov says lines 351 and 352 are uncovered. That's probably not crucial that they're covered, other than someone might come along later and optimize this code and there's nothing protecting the expectation. At the very least, we should have a `# pragma: no cover` comment on line 351 to suppress the issue _and_ signal to a future reviewer that the behavior is uncovered (and thus requires more scrutiny to remove).

```suggestion
            except RuntimeError:  # pragma: no cover
                warnings.warn(
```

Reply:
I'm going to go with the `suppress` approach.